### PR TITLE
feat(RN): Troubleshoot sensitive transaction data

### PR DESCRIPTION
**Pull request by philipphofmann**
## Pre-merge checklist

*If you work at Sentry, you're able to merge your own PR without review, but please don't unless there's a good reason.*

- [x] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## Description of changes

Add to troubleshooting how to fix sending sensitive transaction data.

### CODE REVIEW

File: src/platforms/react-native/troubleshooting/index.mdx [hunk 1]
```diff
@@ -338,3 +338,18 @@ Sentry.init({
   ],
 });
 ```
+
+## Transaction Data Contains Sensitve Information
+
+The SDK might send sensitive data such as username and password for auto-generated transactions before 5.15.2 from navigation routing parameters. You can fix this issue by either updating to [5.15.2](https://github.com/getsentry/sentry-react-native/releases/tag/5.15.2) or higher or dropping the data manually with `beforeSendTransaction`:
```
Comment (kahest) [inline src/platforms/react-native/troubleshooting/index.mdx:344]:
```suggestion
If you use automatic routing instrumentation for React Navigation, sensitive data you put in routing parameters can get sent to Sentry. You can prevent this by either updating the SDK to [5.15.2](https://github.com/getsentry/sentry-react-native/releases/tag/5.15.2) or higher, or dropping the data manually with `beforeSendTransaction`:
```

**Review by buenaflor — APPROVED** [review]
👍

File: src/platforms/react-native/troubleshooting/index.mdx [hunk 2]
```diff
@@ -338,3 +338,18 @@ Sentry.init({
   ],
 });
 ```
+
+## Transaction Data Contains Sensitve Information
```
Comment (kahest) [inline src/platforms/react-native/troubleshooting/index.mdx:342]:
```suggestion
## Routing Transaction Data Contains Sensitve Information
```

**Review by kahest — COMMENTED** [review]
Thanks for adding this 🙏 left wording suggestions

File: src/platforms/react-native/troubleshooting/index.mdx (hunk 2 above)
Comment (markushi) [inline src/platforms/react-native/troubleshooting/index.mdx:342]:
```suggestion
## Transaction Data Contains Sensitive Information
```

**Review by kahest — APPROVED** [review]
looking good 🚀

**Review by lucas-zimerman — APPROVED** [review]

**Pull request merged**